### ISSUE TITLE
Windows: Disconnected network drive causes fern to freeze entire nvim

### ISSUE THREAD

**Issue by sevenc-nanashi** [lambdalisue/vim-fern#501]
On PC with Disconnected Network drive, entire nvim freezes for ~1 minute.

Steps:
- Add a network drive
- Close or shutdown the destination of network drive
- Open Fern

Version info: Neovim / 0.10.0 / Windows 10

**Comment by lambdalisue** [comment]
Let me know the exact command and working directory for opening fern.

**Comment by sevenc-nanashi** [comment]
Command I used is `Fern . -drawer -width=40`, Working directory I used is `E:/pro_psfanmade/MikuMikuWorld`.
And profile:

<details>
<summary>Details</summary>

```
fern#scheme#file#util#list_drives [enter] 
| fern#internal#viewer:init [enter] 
| | fern#helper:helper.async.expand_node [enter] 
| | | fern#internal#node#expand [enter] 
| | | | fern#internal#node#children [enter] 
| | | | | fern#scheme#file#provider:provider_get_children [enter] 
| | | | | | fern#scheme#file#util#list_entries_readdir [enter] 
| | | | | | fern#scheme#file#util#list_entries_readdir [leave] 0.002113 [0.002113]
| | | | | fern#scheme#file#provider:provider_get_children [leave] 0.012520 [0.012520]
| | | | fern#scheme#file#util#list_drives [leave] 0.506302 [0.506302]
| | | fern#internal#node#children [leave] 21.084961 [21.084961]
| | | fern#internal#node#expand [children] 21.085594 [21.085594]
| | | fern#internal#node#expand [sort] 0.001731 [21.087325]
| | | fern#internal#node#expand [extend] 0.000367 [21.087693]
| | fern#internal#node#expand [leave] 0.000332 [21.088025]
| | fern#helper:helper.async.update_nodes [enter] 
| | | fern#internal#core#update_nodes [enter] 
| | | | fern#internal#core#update_nodes [hidden] 0.000898 [0.000898]
| | | | fern#internal#core#update_nodes [include] 0.000691 [0.001589]
| | | | fern#internal#core#update_nodes [exclude] 0.019501 [0.021090]
| | | | fern#internal#core#update_nodes [let] 0.000351 [0.021441]
| | | | fern#internal#core#update_marks [enter] 
| | | | | fern#internal#core#update_marks [resolve] 0.000521 [0.000521]
| | | | | fern#internal#core#update_marks [key] 0.000626 [0.001148]
| | | | | fern#internal#core#update_marks [filter] 0.000460 [0.001608]
| | | | fern#internal#core#update_marks [leave] 0.000331 [0.001939]
| | | fern#internal#core#update_nodes [leave] 0.002569 [0.024009]
| | fern#helper:helper.async.update_nodes [leave] 0.024925 [0.024925]
| fern#helper:helper.async.expand_node [leave] 21.118625 [21.118625]
| fern#internal#viewer:init [expand] 21.356050 [21.356050]
| fern#helper:helper.async.redraw [enter] 
| | fern#renderer#nerdfont#s:render [enter] 
| | fern#renderer#nerdfont#s:render [leave] 0.008054 [0.008054]
| | fern#helper:helper.async.remark [enter] 
| | fern#helper:helper.async.remark [leave] 0.005056 [0.005056]
| | fern_git_status#process#show_toplevel [enter] 
| | | fern_git_status#process#status [enter] 
| | | fern#helper:helper.async.redraw [leave] 0.377535 [0.377535]
| | | fern#internal#viewer:init [redraw] 0.378449 [21.734498]
| | fern#internal#viewer:init [leave] 0.000230 [21.734728]
| fern_git_status#process#show_toplevel [leave] 0.375574 [0.375574]
fern_git_status#process#status [leave] 0.301372 [0.301372]
fern_git_status#investigator#s:complete_directories [enter] 
fern_git_status#investigator#s:complete_directories [leave] 0.001764 [0.001764]
fern_git_status#investigator#s:prepend_toplevel [enter] 
fern_git_status#investigator#s:prepend_toplevel [leave] 0.000715 [0.000715]
fern_git_status#investigator#s:dict_from_entries [enter] 
fern_git_status#investigator#s:dict_from_entries [leave] 0.000748 [0.000748]
fern#helper:helper.async.redraw [enter] 
| fern#renderer#nerdfont#s:render [enter] 
| fern#renderer#nerdfont#s:render [leave] 0.008029 [0.008029]
| fern#helper:helper.async.remark [enter] 
| fern#helper:helper.async.remark [leave] 0.001992 [0.001992]
fern#helper:helper.async.redraw [leave] 0.012547 [0.012547]
```

</details>

**Comment by lambdalisue** [comment]
Unfortunately, I could not reproduce your situation so I'm not sure if this fixes your issue but could you try #504 ?

**Comment by sevenc-nanashi** [comment]
Actually, I removed that drive after opening this issue, so I cannot reproduce the issue now.
